### PR TITLE
document AugmentedMemoizationPolicy

### PR DESCRIPTION
**Pull request by ortsaCniveK**
**Proposed changes**:
- document `AugmentedMemoizationPolicy`
> Created using [rasa source](https://github.com/RasaHQ/rasa/blob/7fcfdd9eaa8704dc44ecbdb2927eb1a07b10d333/rasa/core/policies/memoization.py#L257) and [this](https://forum.rasa.com/t/policies-and-stories-problem/12779/4) thread where @akelad explains the `AugmentedMemoizationPolicy`.

I haven't used .rst files ever, so let me know if there's an issue, but it seemed to be working when I ran `make livedocs`.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by erohmensing** [comment]
Thanks for the PR! We'll give it a review as soon as possible 🙂

**Comment by akelad** [comment]
@Ghostvv can you give this a first review on content?

File: docs/core/policies.rst [hunk 1]
```diff
@@ -417,6 +417,26 @@ training data. It predicts the next action with confidence ``1.0``
...
+.. note::
+  
+  If it is needed to recall turns from training dialogues where some
+  slots might not be set during prediction time, add relevant stories
+  without such slots to training data, e.g. reminder stories. Since 
+  slots that are set some time in the past are preserved in all
+  future feature vectors until they are set to None, this policy has
+  a capability to recall the turns up to ``max_history`` from training
+  stories during prediction even if additional slots were filled in
+  the past for current dialogue.
```
Comment (Ghostvv) [inline docs/core/policies.rst]: Could you please make it a little bit shorter

File: docs/core/policies.rst [hunk 2]
```diff
@@ -417,6 +417,26 @@ training data. It predicts the next action with confidence ``1.0``
 if this exact conversation exists in the training data, otherwise it
 predicts ``None`` with confidence ``0.0``.
 
+Augmented Memoization Policy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``AugmentedMemoizationPolicy`` remembers examples from training
+stories for up to ``max_history`` turns. It has a forgetting mechanism
```
Comment (akelad) [inline docs/core/policies.rst]:
```suggestion
stories for up to ``max_history`` turns, just like the ``MemoizationPolicy``. Additionally, it has a forgetting mechanism
```

File: docs/core/policies.rst (hunk 1 above)
Comment (akelad) [reply docs/core/policies.rst]: I'm gonna give some more specific feedback on that :D

Comment (akelad) [reply docs/core/policies.rst]: I would remove this part, since it essentially just repeats what is said above: 
```
Since 
  slots that are set some time in the past are preserved in all
  future feature vectors until they are set to None, this policy has
  a capability to recall the turns up to ``max_history`` from training
  stories during prediction even if additional slots were filled in
  the past for current dialogue."
```

Comment (akelad) [reply docs/core/policies.rst]: And I would rephrase the other part of the note to this:
```
If you have dialogues where some slots may not be set during prediction time (e.g. when using reminders), make sure to add the relevant stories without slots to your training data as well. 
```

Comment (akelad) [reply docs/core/policies.rst]: @Ghostvv let me know what you think

Comment (Ghostvv) [reply docs/core/policies.rst]: I think it's wise versa: during prediction time, there are more slots than during training. This policy can forget these additional slots and recall training story

Comment (akelad) [reply docs/core/policies.rst]:
```
If it is needed to recall turns from training dialogues where some
  slots might not be set during prediction time, add relevant stories
  without such slots to training data, e.g. reminder stories
```
that seems to indicate something different? this is copied directly out of our source code

Comment (Ghostvv) [reply docs/core/policies.rst]: yes, so `some slots might not be set` in training stories compared to `prediction time` 😅

Comment (Ghostvv) [reply docs/core/policies.rst]: We probably want to rephrase there as well

Comment (akelad) [reply docs/core/policies.rst]: can you rephrase the parts of my description you wanted to rephrase and post here? i'm confused about which wording we're changing now 🙈

Comment (Ghostvv) [reply docs/core/policies.rst]:
```
If you have dialogues where some slots that are set during prediction time might not be set in training stories (e.g. in training stories starting with a reminder not all previous slots are set), make sure to add the relevant stories without slots to your training data as well.
```

**Comment by ortsaCniveK** [comment]
@akelad @Ghostvv Sorry, I've been busy and haven't had a chance to update this PR.  I've made the requested changes. Let me know what you think!
